flink-end-to-end-tests/test-scripts/common_ha.sh

```diff
@@ -59,10 +61,8 @@ function verify_logs() {
     local VERIFY_CHECKPOINTS=$2
 
     # verify that we have no alerts
-    if ! [ `cat ${OUTPUT} | wc -l` -eq 0 ]; then
-        echo "FAILURE: Alerts found at the general purpose job."
-        EXIT_CODE=1
-    fi
+    check_logs_for_non_empty_out_files
+    EXIT_CODE=$(($EXIT_CODE+$?))
```

Comment:
Wouldn't it be better to fail early with the actual exit code instead of summing it up and potentially obfuscate what the actual problem was?

Reply:
I made a small change to again print something if this check fails, restoring the existing state.
I think it's fine to fail late; if a test fails you can end up with something like:
```
<log output>
FAILURE: Alerts found at the general purpose job.
FAILURE: A JM did not take over.
FAILURE: A JM did not execute the job.
One or more tests FAILED.
```
I think this isn't obfuscating anything.